haystack/agents/base.py

```diff
@@ -111,6 +111,120 @@ def _process_result(self, result: Any) -> str:
                 return str(result)
 
 
+class ToolsManager:
+    """
+    The ToolsManager manages tools for an Agent.
+    """
+
+    def __init__(
+        self,
+        tools: Optional[List[Tool]] = None,
+        tool_pattern: str = r'Tool:\s*(\w+)\s*Tool Input:\s*("?)([^"\n]+)\2\s*',
+    ):
+        """
+        :param tools: A list of tools to add to the ToolManager. Each tool must have a unique name.
+        :param tool_pattern: A regular expression pattern that matches the text that the Agent generates to invoke
+            a tool.
+        """
+        self.tools: Dict[str, Tool] = {tool.name: tool for tool in tools} if tools else {}
+        self.tool_pattern = tool_pattern
+        self.callback_manager = Events(("on_tool_start", "on_tool_finish", "on_tool_error"))
+
+    def add_tool(self, tool: Tool):
+        """
+        Add a tool to the Agent. This also updates the PromptTemplate for the Agent's PromptNode with the tool name.
+
+        :param tool: The tool to add to the Agent. Any previously added tool with the same name will be overwritten.
+        Example:
+        `agent.add_tool(
+            Tool(
+                name="Calculator",
+                pipeline_or_node=calculator
+                description="Useful when you need to answer questions about math"
+            )
+        )
+        """
+        self.tools[tool.name] = tool
+
+    def get_tool_names(self) -> str:
+        """
+        Returns a string with the names of all tools registered
+        """
+        return ", ".join(self.tools.keys())
+
+    def get_tools(self) -> List[Tool]:
+        """
+        Returns a list of all tool instances registered
+        """
+        return list(self.tools.values())
+
+    def get_tool_names_with_descriptions(self) -> str:
+        """
+        Returns a string with the names and descriptions of all tools registered
+        """
+        return "\n".join([f"{tool.name}: {tool.description}" for tool in self.tools.values()])
+
+    def has_tools(self) -> bool:
+        """
+        Returns True if this ToolsManager has any tools, False otherwise.
+        """
+        return len(self.tools) > 0
+
+    def has_tool(self, tool_name: str) -> bool:
+        """
+        Returns True if this ToolsManager has a tool with the given name, False otherwise.
+
+        :param tool_name: The name of the tool for which you want to check whether the ToolsManager has it.
+        """
+        return tool_name in self.tools
```

Comment:
I don't think we need all these methods. I would just expose the `tools` dictionary as a property:
```
@property 
def tools(self):
  return self._tools
```
then you can simply do 
```
tm = ToolManager()
print(len(tm.tools))
print("foo" in tm.tools)
```